### PR TITLE
Allow configuring request timeout for InstallRequest

### LINKED ISSUE #12793: Allow configuring request timeout for InstallRequest
**Is your feature request related to a problem? Please describe.**

We can configure a raft request timeout that is applicable to all requests send between raft leaders and followers. InstallRequest send snapshot files which are considerably larger than other raft requests. As a result sometimes it takes longer to send the request and get a response, especially on networks with higher latency between brokers. This can result in timeout exception and the snapshot replication will have to restart from the beginning (See #11496). 

To workaround, we can increase raft request timeout. But increasing this can affect how fast failures are detected, how fast requests are retried etc. So it would be better to be able to set a higher request timeout just of InstallRequest.

**Describe the solution you'd like**

Expose a configuration to set request timeout for InstallRequest. This should be different from the existing raft request timeout configuration.

**Describe alternatives you've considered**

- Fix #11496. This will not solve the issue, but reduces the impact if the request timesout.
- Instead of sending the files as it is, split into to smaller requests. This would be a better solution, but requires more effort.

**Additional context**
Related to https://jira.camunda.com/browse/SUPPORT-16901

### PR DESCRIPTION
**Pull request by rodrigo-lourenco-lopes**
## Description

Allows configuring request timeout for install requests. The default value was set as 10 seconds.

To test the configuration a benchmark was created with the default install request timeout of 500ms and request timeout of 2500ms.
After the creation, it was observed that the benchmark was stable in the Grafana dashboards.
Following this, a latency of 500ms was introduced on the brokers which caused only the install requests timeout to fail, as observed in the Response Failure Count panel in Grafana.

Screenshot 2023-05-30 at 15 07 41

The same Chaos mesh run on a different benchmark without these timeout changes did not reproduce these install request timeout exceptions.

## Related issues

closes #12793

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by deepthidevaki — APPROVED** [review]
:rocket: Thanks Rodrigo.

Please address my comments before merging. No need for another review.

Could you also run a benchmark with chaos mesh and evaluate with a higher install request timeout. 
1. Increase network latency to 35ms, and observe InstallRequests timesout
2. Increase installRequestTimeout to 10s, and observe there are no more timesout. Verify if the cluster is running stable for some time. 
You can do it after merging this PR. Please add a comment to this PR with the results.

**Comment by deepthidevaki** [comment]
@rodrigo-lourenco-lopes Don't forget to backport as we discussed. Backport it to 8.0, 8.1 and 8.2. You can add labels `backport stable/X` for each of the stable branch to backport. Backport action creates backport PRs. Once the PR is created you can approve and merge them. If if failed to create PR you have to manually backport the commits.

**Comment by rodrigo-lourenco-lopes** [comment]
bors retry

**Comment by rodrigo-lourenco-lopes** [comment]
bors retry

**Comment by rodrigo-lourenco-lopes** [comment]
Further tests were conducted:

1st one was a benchmark with both request timeouts and install request timeouts with the value of 2500ms, and adding 35ms of latency:

image

The failures shown above are java.net.ConnectException.

the 2nd test was run with requests timeouts set with 2500ms and install requests timeouts with 10s, and also adding 35ms of latency:

image

The image above shows also java.net.ConnectException exceptions but on a volume roughly 10 times smaller.

**Pull request merged**
